website/docs/best-practices/FileDownloads.md

```diff
@@ -0,0 +1,129 @@
+---
+id: file-download
+title: File Download
+---
+
+When automating file downloads in web testing, it's essential to handle them consistently across different browsers to ensure reliable test execution.
+
+Here, we provide best practices for file downloads and demonstrate how to configure download directories for **Google Chrome**, **Mozilla Firefox**, and **Microsoft Edge**.
+
+## Download Paths
+
+**Hardcoding** download paths in test scripts can lead to maintenance issues and portability problems. Utilize **relative paths** for download directories to ensure portability and compatibility across different environments.
+
+```javascript
+// 👎
+// Hardcoded download path
+const downloadPath = '/path/to/downloads';
+
+// 👍
+// Relative download path
+const downloadPath = path.join(__dirname, 'downloads');
+```
+
+## Wait Strategies
+
+Failing to implement proper wait strategies can lead to race conditions or unreliable tests, especially for download completion. Implement **explicit** wait strategies to wait for file downloads to complete, ensuring synchronization between test steps.
+
+```javascript
+// 👎
+// No explicit wait for download completion
+await browser.pause(5000);
+
+// 👍
+// Wait for file download completion
+await waitUntil(async ()=> await fs.existsSync(downloadPath), 5000);
+```
+
+## Configuring Download Directories
+
+To override file download behavior for **Google Chrome**, **Mozilla Firefox**, and **Microsoft Edge**, provide the download directory in the WebDriverIO capabilities:
+
+<Tabs
+defaultValue="chrome"
+values={[
+{label: 'Chrome', value: 'chrome'},
+{label: 'Firefox', value: 'firefox'},
+{label: 'Microsoft Edge', value: 'edge'},
+]
+}>
+
+<TabItem value='chrome'>
+
+```javascript
+capabilities: {
+    browserName: 'chrome',
+    'goog:chromeOptions': {
+        prefs: {
+            'download.default_directory': downloadPath
+        }
+    }
+}
+```
```

Comment:
````suggestion
```javascript reference title="wdio.conf.js"
https://github.com/webdriverio/example-recipes/blob/84dda93011234d0b2a34ee0cfb3cdfa2a06136a5/testDownloadBehavior/wdio.conf.js#L8-L16
```
````

Reply:
ok. I have added the example recipe link in it. 

if you look at end of these capabilities, I have already added that:
```
For an example implementation, refer to the [ WebdriverIO Test Download Behavior Recipe ] 
( https://github.com/webdriverio/example-recipes/tree/main/testDownloadBehavior ).
```


One more thing, Should i replace the capaiblities with the link? 

If i do so, only the link will appear in it like this..
<img width="1153" alt="Screenshot 2024-02-07 at 11 20 47 AM" src="https://github.com/webdriverio/webdriverio/assets/64587361/a78cb975-9244-401d-9a87-67cf92396019">

Reply:
not `refrence` it's `reference` 😉